lib/src/barback/asset_environment.dart

```diff
@@ -167,21 +165,51 @@ class AssetEnvironment {
   /// go to barback immediately.
   Set<AssetId> _modifiedSources;
 
+  /// The compiler mode for this environment.
+  final Compiler compiler;
+
   AssetEnvironment._(this.graph, this.barback, this.mode, this._watcherType,
-      this._hostname, this._basePort, this.environmentConstants);
+      this._hostname, this._basePort, this.environmentConstants, this.compiler);
 
-  /// Gets the built-in [Transformer]s that should be added to [package].
+  /// Gets the built-in [Transformer]s or [AggregateTransformer]s that should be
+  /// added to [package].
```

Comment:
Can you document why it's a sequence of set? Presumably because the outer sequence is phases?